community/cloud-foundation/templates/ip_reservation/ip_reservation.py.schema

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+info:
+  title: IP Reservation
+  author: Sourced Group Inc.
+  description: Reservers internal, external, or global IP address.
+
+required:
+  - ips
+
+properties:
+  ips:
```

Comment:
I would like to take this into 2 templates.

IP address template: All the logic what is implemented here, except it gets 1 "item" as an input. Properly validates it in the schema, no looping in the template.

IP reservation template: The schema can be exactly what is here with a note to the IP addresses schema. The template itself is only looping through 'ips' and passing each item to the IP Address template.

The goal is to separate the "looping complexity" from the "Multiple type of IP handled uniformly complexity"